frontend/packages/dev-console/src/components/pipelineruns/logs/LogsWrapperComponent.tsx

```diff
@@ -3,18 +3,21 @@ import * as _ from 'lodash';
 import { MultiStreamLogs } from './MultiStreamLogs';
 import { FirehoseResult } from '@console/internal/components/utils';
 import { PodKind } from '@console/internal/module/k8s';
+import { DownloadAll } from './logs-utils';
 
 type LogsWrapperComponentProps = {
   obj?: FirehoseResult<PodKind>;
   taskName: string;
+  downloadAllLabel?: string;
+  onDownloadAll?: () => Promise<DownloadAll>;
 };
 
-const LogsWrapperComponent: React.FC<LogsWrapperComponentProps> = ({ obj, taskName }) => {
+const LogsWrapperComponent: React.FC<LogsWrapperComponentProps> = ({ obj, ...props }) => {
   const ref = React.useRef(obj?.data);
   if (!_.isEmpty(obj?.data)) {
     ref.current = obj.data;
   }
-  return ref.current ? <MultiStreamLogs taskName={taskName} resource={ref.current} /> : null;
+  return ref.current ? <MultiStreamLogs resource={ref.current} {...props} /> : null;
```

Comment:
```suggestion
  return ref.current ? <MultiStreamLogs {...props} resource={ref.current} /> : null;
```
If someone sends in `resource` your wrapper is near useless.

Reply:
`resource` is not a prop of LogsWrapperComponent. There will be type errors if there is a resource prop on LogWrapper.
But made changes as per your suggestion.

Reply:
What type error did you see? I see no type errors if I add `resource?: boolean;` to the `LogWrapperComponentProps`. You'll see an unrelated "you should provide all required props" error if you make it required... is that what you saw?

Reply:
This error 
```
Type '{ taskName: any; downloadAllLabel: string; onDownloadAll: () => Promise<Error>; resource: string; }' is not assignable to type 'IntrinsicAttributes & LogsWrapperComponentProps & { children?: ReactNode; }'.
  Property 'resource' does not exist on type 'IntrinsicAttributes & LogsWrapperComponentProps & { children?: ReactNode; }'.
```

Reply:
Did you just provide a property without changing the type? That's a different-different error lol.

If you type it on your LogWrapper and provide it (or spread another unknown typed object, or make it optional), you're effectively going to remove your ref value without the change I requested above. Spreads will override everything before it (this is true for React AND JavaScript objects). 

I hope you're understanding the merit behind this change. It's important for you to understand how this works so accidental spreads do not happen over other local values you're providing. TypeScript will only help you so much. Please let me know if you need to talk about this @sahil143.